### ISSUE TITLE
`ln` file completion does not work with flags

### ISSUE THREAD

**Issue by zx8** [fish-shell/fish-shell#4089]
`ln -<flag> <TAB>` is not working. Without flags, it works fine. Not sure if it's related to https://github.com/fish-shell/fish-shell/issues/4061?

```
$ fish --version
fish, version 2.5.0-419-g9236e27e
```

**Expected**
```sh
$ ln -s ~/<TAB>
# list of matching directories
```

**Actual**
```sh
$ ln -s ~/<TAB>
# nothing
```

**Comment by zx8** [comment]
> Not sure if it's related to https://github.com/fish-shell/fish-shell/issues/4061#issuecomment-304304560

Seems to be, also bisects to https://github.com/fish-shell/fish-shell/commit/a3a069234b92e86e204d58fbc7d3ddaebafe2cc7.

**Comment by faho** [comment]
>Seems to be, also bisects to a3a0692.

Yes, it's related. The issue is that `ln`s completions are bogus - they just have "-f" everywhere.

#4090 passes my smell-test, but I won't have the time and sobriety to merge it until next week and github's diff is weird (one block of removals followed by one block of additions) which makes it hard to see if a line was accidentally dropped or similar.

**Comment by krader1961** [comment]
Fixed by commit 5e39efa45 from George Christou.
